Content.Server/Xenoarchaeology/XenoArtifacts/Effects/Systems/ChemicalPuddleArtifactSystem.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Content.Server.Fluids.EntitySystems;
+using Content.Server.Xenoarchaeology.XenoArtifacts.Effects.Components;
+using Content.Server.Xenoarchaeology.XenoArtifacts.Events;
+using Robust.Shared.Random;
+
+namespace Content.Server.Xenoarchaeology.XenoArtifacts.Effects.Systems;
+
+/// <summary>
+/// This handles <see cref="ChemicalPuddleArtifactComponent"/>
+/// </summary>
+public sealed class ChemicalPuddleArtifactSystem : EntitySystem
+{
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly ArtifactSystem _artifact = default!;
+    [Dependency] private readonly SpillableSystem _spillable = default!;
+
+    /// <summary>
+    /// The key for the node data entry containing
+    /// the chemicals that the puddle is made of.
+    /// </summary>
+    public const string NodeDataChemicalList = "nodeDataSpawnAmount";
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<ChemicalPuddleArtifactComponent, ArtifactActivatedEvent>(OnActivated);
+    }
+
+    private void OnActivated(EntityUid uid, ChemicalPuddleArtifactComponent component, ArtifactActivatedEvent args)
+    {
+        if (!TryComp<ArtifactComponent>(uid, out var artifact))
+            return;
```

Comment:
can ArtifactActivatedEvent ever trigger on something without the artifact component?

Reply:
no, but i'm just getting the component here to save a possible extra resolve.